Description: Add pkg-config and cmake scripts
Origin: vendor
Forwarded: https://github.com/redis/hiredis/pull/129
Author: Tom Lee <debian@tomlee.co>
Last-Update: 2012-11-06

--- /dev/null
+++ b/FindHiredis.cmake
@@ -0,0 +1,34 @@
+# FindHiredis.cmake - Try to find the Hiredis library
+# Once done this will define
+#
+#  HIREDIS_FOUND - System has Hiredis
+#  HIREDIS_INCLUDE_DIR - The Hiredis include directory
+#  HIREDIS_LIBRARIES - The libraries needed to use Hiredis
+#  HIREDIS_DEFINITIONS - Compiler switches required for using Hiredis
+
+
+# use pkg-config to get the directories and then use these values
+# in the FIND_PATH() and FIND_RARY() calls
+FIND_PACKAGE(PkgConfig)
+PKG_SEARCH_MODULE(PC_HIREDIS REQUIRED hiredis)
+
+SET(HIREDIS_DEFINITIONS ${PC_HIREDIS_CFLAGS_OTHER})
+
+FIND_PATH(HIREDIS_INCLUDE_DIR NAMES hiredis/hiredis.h
+   HINTS
+   ${PC_HIREDIS_INCLUDEDIR}
+   ${PC_HIREDIS_INCLUDE_DIRS}
+   PATH_SUFFIXES hiredis
+   )
+
+FIND_LIBRARY(HIREDIS_LIBRARIES NAMES hiredis
+   HINTS
+   ${PC_HIREDIS_DIR}
+   ${PC_HIREDIS_LIBRARY_DIRS}
+   )
+
+
+INCLUDE(FindPackageHandleStandardArgs)
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(Hiredis DEFAULT_MSG HIREDIS_LIBRARIES HIREDIS_INCLUDE_DIR)
+
+MARK_AS_ADVANCED(HIREDIS_INCLUDE_DIR HIREDIS_LIBRARIES)
--- a/Makefile
+++ b/Makefile
@@ -8,19 +8,23 @@
 TESTS=hiredis-test
 LIBNAME=libhiredis
 PKGCONFNAME=hiredis.pc
+CMAKENAME=FindHiredis.cmake
 
 HIREDIS_MAJOR=$(shell grep HIREDIS_MAJOR hiredis.h | awk '{print $$3}')
 HIREDIS_MINOR=$(shell grep HIREDIS_MINOR hiredis.h | awk '{print $$3}')
 HIREDIS_PATCH=$(shell grep HIREDIS_PATCH hiredis.h | awk '{print $$3}')
 
 # Installation related variables and target
-PREFIX?=$(DESTDIR)/usr
+INSTALL_PREFIX?=$(DESTDIR)/usr
+PREFIX?=$(INSTALL_PREFIX)
 INCLUDE_PATH?=include/hiredis
 LIBRARY_PATH?=lib
 PKGCONF_PATH?=pkgconfig
-INSTALL_INCLUDE_PATH= $(DESTDIR)$(PREFIX)/$(INCLUDE_PATH)
-INSTALL_LIBRARY_PATH= $(DESTDIR)$(PREFIX)/$(LIBRARY_PATH)
+CMAKE_MODULES_PATH?=share/cmake/Modules
+INSTALL_INCLUDE_PATH= $(INSTALL_PREFIX)/$(INCLUDE_PATH)
+INSTALL_LIBRARY_PATH= $(INSTALL_PREFIX)/$(LIBRARY_PATH)
 INSTALL_PKGCONF_PATH= $(INSTALL_LIBRARY_PATH)/$(PKGCONF_PATH)
+INSTALL_CMAKE_PATH= $(INSTALL_PREFIX)/$(CMAKE_MODULES_PATH)
 
 # redis-server configuration used for testing
 REDIS_PORT=56379
@@ -160,7 +164,7 @@
 	@echo Libs: -L\$${libdir} -lhiredis >> $@
 	@echo Cflags: -I\$${includedir} -D_FILE_OFFSET_BITS=64 >> $@
 
-install: $(DYLIBNAME) $(STLIBNAME) $(PKGCONFNAME)
+install: $(DYLIBNAME) $(STLIBNAME) $(PKGCONFNAME) $(CMAKENAME)
 	mkdir -p $(INSTALL_INCLUDE_PATH) $(INSTALL_LIBRARY_PATH)
 	$(INSTALL) hiredis.h async.h read.h sds.h adapters $(INSTALL_INCLUDE_PATH)
 	$(INSTALL) $(DYLIBNAME) $(INSTALL_LIBRARY_PATH)/$(DYLIB_MINOR_NAME)
@@ -169,6 +173,8 @@
 	$(INSTALL) $(STLIBNAME) $(INSTALL_LIBRARY_PATH)
 	mkdir -p $(INSTALL_PKGCONF_PATH)
 	$(INSTALL) $(PKGCONFNAME) $(INSTALL_PKGCONF_PATH)
+	mkdir -p $(INSTALL_CMAKE_PATH)
+	$(INSTALL) $(CMAKENAME) $(INSTALL_CMAKE_PATH)
 
 32bit:
 	@echo ""
@@ -195,4 +201,4 @@
 noopt:
 	$(MAKE) OPTIMIZATION=""
 
-.PHONY: all test check clean dep install 32bit gprof gcov noopt
+.PHONY: all test check clean dep install 32bit gprof gcov noopt hiredis.pc
--- /dev/null
+++ b/hiredis.pc.in
@@ -0,0 +1,11 @@
+prefix=@PREFIX@
+exec_prefix=${prefix}
+libdir=@LIB_PREFIX@
+includedir=${prefix}/include
+
+Name: Hiredis
+Description: Minimalistic C client library for the Redis database.
+Version: 0.11.0
+Libs: -L${libdir} -lhiredis
+Cflags: -I${includedir}/hiredis @CFLAGS@
+
